libres/lib/config/config_schema_item.cpp

```diff
@@ -66,398 +66,465 @@ typedef struct validate_struct validate_type;
-        else
-          error_message = util_alloc_sprintf("Error:: Keyword:%s must have at least %d arguments.",item->kw , item->validate->argc_min);
-
-        config_error_add( error_list , error_message );
-      }
+bool config_schema_item_validate_set(const config_schema_item_type *item,
+                                     stringlist_type *token_list,
+                                     const char *config_file,
+                                     const config_path_elm_type *path_elm,
+                                     config_error_type *error_list) {
+    bool OK = true;
+    int argc = stringlist_get_size(token_list) - 1;
+    if (item->validate->argc_min >= 0) {
+        if (argc < item->validate->argc_min) {
+            OK = false;
+            {
+                char *error_message;
+                if (config_file != NULL)
+                    error_message = util_alloc_sprintf(
+                        "Error when parsing config_file:\"%s\" Keyword:%s must "
+                        "have at least %d arguments.",
+                        config_file, item->kw, item->validate->argc_min);
+                else
+                    error_message = util_alloc_sprintf(
+                        "Error:: Keyword:%s must have at least %d arguments.",
+                        item->kw, item->validate->argc_min);
+
+                config_error_add(error_list, error_message);
+            }
+        }
     }
-  }
 
-  if (item->validate->argc_max >= 0) {
-    if (argc > item->validate->argc_max) {
-      OK = false;
-      {
-        char * error_message;
-
-        if (config_file != NULL)
-          error_message = util_alloc_sprintf("Error when parsing config_file:\"%s\" Keyword:%s must have maximum %d arguments.",config_file , item->kw , item->validate->argc_max);
-        else
-          error_message = util_alloc_sprintf("Error:: Keyword:%s must have maximum %d arguments.",item->kw , item->validate->argc_max);
-
-        config_error_add( error_list , error_message );
-      }
+    if (item->validate->argc_max >= 0) {
+        if (argc > item->validate->argc_max) {
+            OK = false;
+            {
+                char *error_message;
+
+                if (config_file != NULL)
+                    error_message = util_alloc_sprintf(
+                        "Error when parsing config_file:\"%s\" Keyword:%s must "
+                        "have maximum %d arguments.",
+                        config_file, item->kw, item->validate->argc_max);
+                else
+                    error_message = util_alloc_sprintf(
+                        "Error:: Keyword:%s must have maximum %d arguments.",
+                        item->kw, item->validate->argc_max);
+
+                config_error_add(error_list, error_message);
+            }
+        }
     }
-  }
 
-  /*
+    /*
```

Comment:
Indentation of comment now looks weird because the indentation of the code has changed.